### PR TITLE
refactor(loggers): refine types

### PR DESCRIPTION
**Pull request by emmenko**

### CODE REVIEW

File: packages-backend/loggers/src/formatters/rewrite-fields.ts [hunk 1]
```diff
@@ -8,29 +8,25 @@ import unsetIn from 'lodash/unset';
 type TRewriteField = {
   from: string;
   to: string;
-  replaceValue?: (value: string) => unknown;
+  replaceValue?: (value: unknown) => unknown;
```
Comment (emmenko) [inline packages-backend/loggers/src/formatters/rewrite-fields.ts:11]: This is not always a string.

File: packages-backend/loggers/src/formatters/rewrite-fields.ts [hunk 2]
```diff
@@ -8,29 +8,25 @@ import unsetIn from 'lodash/unset';
...
   if (val) {
-    unsetIn(info, jsonPath);
-    setIn(info, newJsonPath, replaceValue ? replaceValue(val) : val);
+    unsetIn(info, field.from);
+    setIn(info, field.to, field.replaceValue ? field.replaceValue(val) : val);
   }
 }
 
-const rewriteFieldsFormatter = format(
+const rewriteFieldsFormatter: TCreateFormatter = format(
```
Comment (emmenko) [inline packages-backend/loggers/src/formatters/rewrite-fields.ts:26]: The format wrapper has the options argument typed as `any`, so we explicitly type the function with our options type.

image

File: packages-backend/loggers/src/formatters/rewrite-fields.ts (hunk 1 above)
Comment (emmenko) [reply packages-backend/loggers/src/formatters/rewrite-fields.ts:11]: I might have find a way to use a generic...

File: packages-backend/loggers/src/formatters/rewrite-fields.ts (hunk 2 above)
Comment (tdeekens) [reply packages-backend/loggers/src/formatters/rewrite-fields.ts:26]: Nice catch!

**Comment by emmenko** [comment]
Putting on draft as I'm still working on it.

File: packages-backend/loggers/src/formatters/rewrite-fields.ts [hunk 3]
```diff
@@ -5,35 +5,35 @@ import getIn from 'lodash/get';
 import setIn from 'lodash/set';
 import unsetIn from 'lodash/unset';
 
-type TRewriteField = {
+type TRewriteField<FieldValue> = {
   from: string;
   to: string;
-  replaceValue?: (value: string) => unknown;
+  replaceValue?: (value: FieldValue) => unknown;
```
Comment (emmenko) [inline packages-backend/loggers/src/formatters/rewrite-fields.ts:11]: The field can now be typed using a generic.

Comment (adnasa) [reply packages-backend/loggers/src/formatters/rewrite-fields.ts:11]: nice!

**Pull request merged**